### DISCUSSION TITLE
How to properly use `require` in user scripts

### DISCUSSION THREAD

**Discussion by LeonardoGentile** [SilentVoid13/Templater#1339] [Help] [answered]
Hello, 
I wanted to split my usser scripts in reusable modules, but it seems I can't make `require` works.
After digging into the issues of the repo I've found [this](https://github.com/SilentVoid13/Templater/issues/539) but I'm still unsure how this should work.

I've simply tried the relative path require with `const script_2 = require('./script_2.js')` but it doesn't work. I've also tried some absolute path combination but still no luck. 

Then at the end of that discussion [someone suggested](https://github.com/SilentVoid13/Templater/issues/539#issuecomment-1343155111) to pass in `tp` into the function that needs to require the other module and use it like `tp.user.script_2;` 

Can someone clarify how to properly uses `require` from within a user script?
Also, can I put my common code to import into another folder, still within the obsidian vault but outside templater user scripts folder?

**Comment by Zachatoo** [comment, marked as answer]
Referencing scripts on `tp.user` is how I would do it if you're just using them for Templater.

You may want to look into other plugins for truly shared code, [Modules](https://github.com/polyipseity/obsidian-modules) might do the trick.
